src/bio/bio_wal.c

```diff
@@ -1932,6 +1928,22 @@ wal_open(struct bio_meta_context *mc)
 	if (rc)
 		return rc;
 
+	/* Auto upgrade WAL header from version 1 to 2 */
+	if (hdr->wh_version == 1) {
+		struct wal_header_v1 *v1 = (struct wal_header_v1 *)hdr;
+
+		v1->wh_csum     = 0;
+		hdr->wh_version = BIO_WAL_VERSION;
+		uuid_copy(hdr->wh_pool_id, mc->mc_wal->bic_pool_id);
+		hdr->wh_vos_id = mc->mc_meta_hdr.mh_vos_id;
+
+		rc = write_header(mc, mc->mc_wal, hdr, sizeof(*hdr), &hdr->wh_csum);
```

Comment:
Is there anything placed directly after the WAL header in the blob in v1?

Reply:
No, the first 4k block is used for header.